swap/src/protocol/alice/encrypted_signature.rs

```diff
@@ -59,48 +48,42 @@ impl Default for Behaviour {
 
         Self {
             rr: RequestResponse::new(
-                OneShotCodec::default(),
+                CborCodec::default(),
                 vec![(EncryptedSignatureProtocol, ProtocolSupport::Inbound)],
                 config,
             ),
-            events: Default::default(),
         }
     }
 }
 
-impl NetworkBehaviourEventProcess<RequestResponseEvent<Request, Response>> for Behaviour {
-    fn inject_event(&mut self, event: RequestResponseEvent<Request, Response>) {
+impl From<RequestResponseEvent<EncryptedSignature, ()>> for OutEvent {
+    fn from(event: RequestResponseEvent<EncryptedSignature, ()>) -> Self {
         match event {
             RequestResponseEvent::Message {
+                peer,
                 message:
                     RequestResponseMessage::Request {
                         request, channel, ..
                     },
                 ..
             } => {
-                if let Request::EncryptedSignature(msg) = request {
-                    debug!("Received encrypted signature");
-                    self.events.push_back(OutEvent::Msg(*msg));
-                    // Send back empty response so that the request/response protocol completes.
-                    if let Err(error) = self.rr.send_response(channel, Response::EncryptedSignature)
-                    {
-                        error!("Failed to send Encrypted Signature ack: {:?}", error);
-                    }
+                debug!("Received encrypted signature from {}", peer);
```

Comment:
❓ We don't have to use `push_back` to bubble the event up anymore? Is this because of using `event_process = false` ?

Reply:
I confused myself - the event is returned now.